crypto/fipsmodule/ec/ec_key.c

```diff
@@ -292,38 +293,42 @@ void EC_KEY_set_conv_form(EC_KEY *key, point_conversion_form_t cform) {
   }
 }
 
-int EC_KEY_check_key(const EC_KEY *eckey) {
-  if (!eckey || !eckey->group || !eckey->pub_key) {
-    OPENSSL_PUT_ERROR(EC, ERR_R_PASSED_NULL_PARAMETER);
-    return 0;
-  }
+// ec_key_gen_pct computes the PCT: SP 800-56Arev3 Section 5.6.2.1.4 option ‘b’.
+// Should only be used for NIST P-curves.
+static int ec_key_gen_pct(const EC_KEY *key) {
```

Comment:
seems like the function returns true when `key->priv_key == NULL`, is that intended?